test/test_intensify/test_parallel_successive_halving.py

```diff
@@ -0,0 +1,391 @@
+        for i in range(100):
+            try:
+                intent, run_info = sh.get_next_run(
+                    challengers=challengers,
+                    incumbent=None, chooser=None, run_history=rh)
+            except ValueError as e:
+                # Get configurations until you run out of them
+                print(e)
+                break
+
+            # Regenerate challenger list
+            challengers = [c for c in challengers if c != run_info.config]
+
+            if intent == RunInfoIntent.WAIT:
+                break
+
+            result = target_from_run_info(run_info)
+            rh.add(
+                config=run_info.config,
+                cost=result.cost,
+                time=result.time,
+                status=result.status,
+                instance_id=run_info.instance,
+                seed=run_info.seed,
+                budget=run_info.budget,
+            )
+            incumbent, inc_perf = sh.process_results(
+                run_info=run_info,
+                incumbent=incumbent,
+                run_history=rh,
+                time_bound=100.0,
+                result=result,
+                log_traj=False,
+            )
+        return incumbent, inc_perf
+
+    def test_parallel_same_as_serial_SH(self):
+        """Makes sure we behave the same as a serial run at the end"""
+
+        # Get the run_history for a SH run:
+        rh = RunHistory()
+        stats = Stats(scenario=self.scen)
+        stats.start_timing()
+        SH = SuccessiveHalving(
+            stats=stats,
+            traj_logger=TrajLogger(output_dir=None, stats=stats),
+            rng=np.random.RandomState(12345),
+            deterministic=False,
+            run_obj_time=False,
+            instances=[1, 2, 3, 4, 5],
+            n_seeds=2,
+            initial_budget=2,
+            max_budget=5,
+            eta=2,
+        )
+        incumbent, inc_perf = self._exhaust_run_and_get_incumbent(SH, rh)
+
+        # Just to make sure nothing has changed form SH side to make
+        # this check invalid:
+        # We add config values, so config 4 with 0 and 0 should be the lesser cost
+        self.assertEqual(incumbent, self.config3)
+        self.assertEqual(inc_perf, 7.0)
+
+        # Do the same for PSH, but have multiple SH in there
+        # _add_new_SH returns true if it was able to add a new SH
+        # scheduling is defaulted to serial
+        self.assertTrue(self.PSH._add_new_SH())
+        incumbent_psh, inc_perf_psh = self._exhaust_run_and_get_incumbent(self.PSH, self.rh)
+        self.assertEqual(incumbent, incumbent_psh)
+
+        # This makes sure there is a single incumbent in PSH
+        self.assertEqual(inc_perf, inc_perf_psh)
+
+        # We don't want to loose any configuration, and particularly
+        # we want to make sure the values of SH to PSH match
+        self.assertEqual(len(self.rh.data), len(rh.data))
+
+        configs_sh_rh = {}
+        for k, v in rh.data.items():
```

Comment:
Shouldn't one of these two for loops go over `self.rh` instead of `rh`?

Reply:
thanks very much for catching this. The check was useless because of this. I fixed it.